File: Shared/java/code/threads/ThreadCoordTest.java
class ThreadCoordTest{

	private static int value = 0;

	private static void consume(){
		System.out.printf("Consumer ready...%n");
		System.out.printf("Processed value = %d%n", value * value);
	}

	private static void produce(){
		System.out.printf("Producer ready...%n");
		int result = Worker.doWork();
		value = result;
	}

	public static void main(String[] args) throws Exception{
		produce();
		consume();
	}
}

File: Shared/dotnet/code/threads/threadcoordtest.cs
using System;

static class Program
{
	private static int value = 0;
	
	private static void Consume()
	{
		Console.WriteLine("Consumer ready...");

		Console.WriteLine("Processed value = {0}", value * value);
	}

	private static void Produce()
	{
		Console.WriteLine("Producer ready...");

		int result = Worker.DoWork();

		value = result;

	}

	public static void Main()
	{
		Produce();
	
		Consume();
	}
}	